torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2141,7 +2147,7 @@ Tensor qr_backward(const std::vector<torch::autograd::Variable> &grads, const Te
   auto n = self.size(-1);
 
   TORCH_CHECK(
-      ((m <= n && (!some)) || some),
+      ((m <= n && (!reduced)) || reduced),
       "The derivative is not implemented when nrows > ncols and complete QR. ");
```

Comment:
Let's improve this error message while we're here. The phrase "and complete QR" is odd and vague. Can we relate it more directly to the modes?

Reply:
done by dc24806a8a. I also added a test and made the two error messages more consistent